x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/LookupJoinExec.java

```diff
@@ -93,9 +93,9 @@ public List<Attribute> addedFields() {
     public List<Attribute> output() {
         if (lazyOutput == null) {
             lazyOutput = new ArrayList<>(left().output());
-            for (Attribute attr : addedFields) {
-                lazyOutput.add(attr);
-            }
+            var addedFieldsNames = addedFields.stream().map(Attribute::name).toList();
+            lazyOutput.removeIf(a -> addedFieldsNames.contains(a.name()));
+            lazyOutput.addAll(addedFields);
```

Comment:
Oh, this is very interesting. Thanks for catching this.

I'm wondering if enforcing unique names for `PhysicalPlan` outputs is actually the right way to go. It is what we assumed so far, so I don't wanna depart from here, but there's also a reason to be more lenient:

In PhysicalPlan land, this output method was _technically_ correct, because it represented exactly what the physical operators will do: they're gonna append the blocks for the added fields to the pages that are processed. The blocks corresponding to shadowed attributes/channels will _not_ be stripped from incoming pages.

This means that the output layout technically has duplicate attribute names. This change just hides them and makes them inaccessible during the physical planning stage.

FWIW, `EvalExec` has the exact same situation. `Eval` performs shadowing, but `EvalOperator` can only ever append blocks, so the output pages actually still contain the blocks corresponding to shadowed attributes.